compiler/rustc_mir/src/const_eval/mod.rs

```diff
@@ -123,8 +123,22 @@ fn const_to_valtree<'tcx>(
                         branches(ecx, n.try_into().unwrap(), None, &mplace)
                     };
                     match mplace.layout.ty.kind() {
-                        // str slices are encoded as a `u8` array.
-                        ty::Str => array(ecx.tcx.types.u8),
+                        ty::Str => {
+                            let n = scalar.to_machine_usize(ecx).unwrap();
+                            if n > 0 {
+                                let ptr = mplace.ptr.assert_ptr();
+                                let s = ecx.memory.get_raw(ptr.alloc_id).unwrap().get_bytes(
+                                    ecx,
+                                    ptr,
+                                    Size::from_bytes(n),
+                                ).unwrap();
+                                let s = std::str::from_utf8(s).unwrap();
+                                let s = Symbol::intern(s);
+                                Ok(Some(ty::ValTree::Str(s)))
+                            } else {
+                                Ok(Some(ty::ValTree::Str(Symbol::intern(""))))
```

Comment:
Can it be `kw::Empty` instead interning empty string?

Reply:
I wasn't sure and went for the quick option. I'll have to look into what happens when interning the empty string.